nemo/collections/tts/data/datalayers.py

```diff
@@ -515,3 +520,222 @@ def preprocess_linear_specs_dataset(valid_filelist, train_filelist, n_fft, hop_l
             raise EOFError("Dataset initialization failed. No files to preprocess validation dataset")
 
     return tar_dir
+
+
+class FastSpeech2Dataset(Dataset):
+    """
+    (Assumes supp data is in energies/ and phoneme_durations/, etc.)
+    """
+
+    # TODO: docstring
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        """Returns definitions of module output ports."""
+        return {
+            'audio_signal': NeuralType(('B', 'T'), AudioSignal()),
+            'a_sig_length': NeuralType(('B'), LengthsType()),
+            'transcripts': NeuralType(('B', 'T'), TokenIndex()),
+            'transcript_length': NeuralType(('B'), LengthsType()),
+            'durations': NeuralType(('B', 'T'), TokenDurationType()),
+            'pitches': NeuralType(('B', 'T'), RegressionValuesType()),
+            'energies': NeuralType(('B', 'T'), RegressionValuesType()),
```

Comment:
What is regressionvaluestype?

Reply:
From the docstring of the type: "Element type to represent labels for regression task." I picked this to match with the pitch prediction type of FastPitch.